### PR TITLE
[Fixes AB#434] Spinner added

### PR DESCRIPTION
**Pull request by Skafec**
Added spinner when project wallet is being initiated with Arkane

### CODE REVIEW

File: src/app/manage-projects/manage-single-project/manage-single-project.component.ts [hunk 1]
```diff
@@ -67,6 +67,7 @@ export class ManageSingleProjectComponent implements OnInit {
 
     createProjectWallet() {
         this.walletService.createProjectWalletTransaction(this.project.uuid).subscribe(res => {
+            SpinnerUtil.showSpinner();
```
Comment (mightymatth) [inline src/app/manage-projects/manage-single-project/manage-single-project.component.ts:70]: This way, after clicking the button to create a project wallet, there will be no spinner until our backend returns response of `this.walletService.createProjectWalletTransaction(this.project.uuid)`. This is an asynchronous method.
Set this line at the beginning of `createProjectWallet()` method (between lines 68 and 69).

**Comment by Skafec** [comment]
Update: In order for Spinner to show when it is added at the start of method, I had to move hideSpinner method out of error handlers. Because we have a async method, code would continue to run and it would get to hideSpinner method inside of fetchAllData too soon.

File: src/app/manage-projects/manage-single-project/manage-single-project.component.ts [hunk 2]
```diff
@@ -66,12 +66,13 @@ export class ManageSingleProjectComponent implements OnInit, AfterViewInit {
                 } else {
                     displayBackendError(err);
                 }
-                SpinnerUtil.hideSpinner();
             });
+            SpinnerUtil.hideSpinner();
```
Comment (mightymatth) [inline src/app/manage-projects/manage-single-project/manage-single-project.component.ts:70]: This line doesn't make sense. subscription from line 57 to 69 represents asynchronous call
Think logically. This here means that you:

1. show spinner, 
2. synchronously start async call (finishes immediately), 
3. hide spinner. 

Practically, this spinner won't be even visible. If it fixes closing some orphan spinner from somewhere else, it's unintentional. Fix closing the right spinner on the place where it should be closed.

**Review by mightymatth — CHANGES_REQUESTED** [review]

**Comment by Skafec** [comment]
Oh so that's why we had unnecessary spinner showing when project had active wallet.

Now fetchAllData() method is cleaned up from spinners and all the spinners are being closed on other
function calls inside it.

File: src/app/manage-projects/manage-single-project/manage-single-project.component.ts [hunk 3]
```diff
@@ -52,10 +52,7 @@ export class ManageSingleProjectComponent implements OnInit, AfterViewInit {
 
     fetchAllData() {
         this.getProject(() => {
-            SpinnerUtil.showSpinner();
```
Comment (mightymatth) [inline src/app/manage-projects/manage-single-project/manage-single-project.component.ts:55]: I didn't mean that spinners should be removed from `fetchAllData()`, but should be set properly: 
1. on the entrance of the function
2. in every possible callback where it should be closed (all callbacks that finishes data fetch; success and error callbacks).

File: src/app/manage-projects/manage-single-project/manage-single-project.component.ts [hunk 4]
```diff
@@ -66,12 +63,12 @@ export class ManageSingleProjectComponent implements OnInit, AfterViewInit {
                 } else {
                     displayBackendError(err);
```
Comment (mightymatth) [inline src/app/manage-projects/manage-single-project/manage-single-project.component.ts:64]: use `hideSpinnerAndDisplayError()` and remove `hideSpinner()` from below.

**Comment by mightymatth** [comment]
Will be fixed in another branch

**Pull request closed without merging**